snmp/datadog_checks/snmp/snmp.py

```diff
@@ -290,167 +345,77 @@ def check_table(
 
-    def parse_metrics(self, metrics, enforce_constraints):
-        if not metrics:
-            raise Exception('Metrics list must contain at least one metric')
-        raw_oids = []
-        table_oids = []
-        mibs_to_load = set()
-        # Check the metrics completely defined
-        for metric in metrics:
-            if 'MIB' in metric:
-                if not ("table" in metric or "symbol" in metric):
-                    raise Exception("When specifying a MIB, you must specify either table or symbol")
-                if not enforce_constraints:
-                    # We need this only if we don't enforce constraints to be able to lookup MIBs manually
-                    mibs_to_load.add(metric["MIB"])
-                if "symbol" in metric:
-                    to_query = metric["symbol"]
-                    try:
-                        table_oids.append(hlapi.ObjectType(hlapi.ObjectIdentity(metric["MIB"], to_query)))
-                    except Exception as e:
-                        self.warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
-                else:
-                    if "symbols" not in metric:
-                        raise Exception("When specifying a table, you must specify a list of symbols")
-                    for symbol in metric["symbols"]:
-                        try:
-                            table_oids.append(hlapi.ObjectType(hlapi.ObjectIdentity(metric["MIB"], symbol)))
-                        except Exception as e:
-                            self.warning("Can't generate MIB object for variable : %s\nException: %s", metric, e)
-                    if "metric_tags" in metric:
-                        for metric_tag in metric["metric_tags"]:
-                            if not ("tag" in metric_tag and ("index" in metric_tag or "column" in metric_tag)):
-                                raise Exception(
-                                    "When specifying metric tags, you must specify a tag, and an index or column"
-                                )
-                            if "column" in metric_tag:
-                                # In case it's a column, we need to query it as well
-                                try:
-                                    table_oids.append(
-                                        hlapi.ObjectType(hlapi.ObjectIdentity(metric["MIB"], metric_tag.get("column")))
-                                    )
-                                except Exception as e:
-                                    self.warning(
-                                        "Can't generate MIB object for variable : %s\nException: %s", metric, e
-                                    )
+    def check(self, instance):
+        """
+        Perform two series of SNMP requests, one for all that have MIB associated
+        and should be looked up and one for those specified by oids.
+        """
+        config = self._load_conf(instance)
```

Comment:
```suggestion
        config = self._conf[instance['name']]
```

Any reason not just using `config = self._conf[instance['name']]` since confs are already loaded ?

**Corollary:**
If `self._load_conf(...)` is only called once, this protection may not be necessary: https://github.com/DataDog/integrations-core/pull/4160/files#diff-c232c5e7cec47d95c2ccdffda2acf33bR234

Reply:
Yeah I didn't want to change all the tests, but ended up doing so.